iosApp/iosApp/Pages/StopDetails/StopDetailsPage.swift

```diff
@@ -134,6 +149,11 @@ struct StopDetailsPage: View {
     }
 
     func changeStop(_ stop: Stop) {
+        leavePredictions()
+        fetchStopData(stop)
```

Comment:
I could also update the call to `connect` in the repository to leave the channel first. We haven't consistently taken that approach, but it might help to standardize on it.

Reply:
Are there any downsides to doing this? It seems like it makes sense for the repo to handle it.

Reply:
I guess that is the approach that the predictions V2 channel takes, so changing it for V1 doesn't really matter, depending on when this change lands relative to the V2 cutover.

Reply:
Ahh, gotcha, yeah if it's not relevant for v2 then I wouldn't bother.